File: src/test/java/com/znsio/teswiz/screen/ScreenShotScreen.java
package com.znsio.teswiz.screen;


public abstract class ScreenShotScreen {

    public static ScreenShotScreen get() {
        return com.znsio.teswiz.screen.ScreenRegistry.getScreen(ScreenShotScreen.class);
    }

    public abstract ScreenShotScreen takeScreenshot();
}

File: src/test/resources/playwright/screens/screen-shot.screen.ts
import type { ScreenContext } from "./screen-context.ts";
import { stayOnCurrentScreen } from "./screen-route.ts";

const PAGE_URL = "https://github.com/znsio/teswiz";

export async function takeScreenshot(screen: ScreenContext): Promise<object> {
  await screen.page.goto(PAGE_URL, { waitUntil: "load" });
  return stayOnCurrentScreen();
}
